cid/export.py

```diff
@@ -96,109 +139,217 @@ def export_analysis(qs):
+        raise CidCritical(f'CID only supports one database. Multiple used: {all_databases}')
 
-    template_id = get_parameter(
-        'template-id',
-        message='Enter template id',
-        default=analysis.get('Name').replace(' ', '-')
-    )
+    if all_databases:
+        athena.DatabaseName = all_databases[0]
 
-    template_version_description = get_parameter(
-        'template-version-description',
-        message='Enter version description',
-        default='vX.X.X' # FIXME: get version from analysis / template
-    )
+    cid_print(f'Analyzing Athena Views: {all_views}. Can take some time.')
+    all_views_data = athena.process_views(all_views)
+    cid_print(f'List of views: {str(list(all_views_data.keys()))}')
 
-    logger.info('Updating template')
-    params = {
-        "AwsAccountId": qs.account_id,
-        "TemplateId": template_id,
-        "Name": template_id,
-        "SourceEntity": {
-            "SourceAnalysis": {
-                "Arn": analysis.get("Arn"),
-                "DataSetReferences": dataset_references
-            }
-        },
-        "VersionDescription": template_version_description,
-    }
-    logger.debug(f'Template params = {params}')
-    try:
-        res = qs.client.update_template(**params)
-        logger.info(f'Template {template_id} updated from Analysis {analysis.get("Arn")}')
-    except qs.client.exceptions.ResourceNotFoundException:
-        res = qs.client.create_template(**params)
-        logger.info(f'Template {template_id} created from Analysis {analysis.get("Arn")}')
-
-    if res['CreationStatus'] not in ['CREATION_IN_PROGRESS', 'UPDATE_IN_PROGRESS']:
-        raise CidCritical(f'failed template operation {res}')
+    # Post processing of views:
+    # - Special treatment for CUR: replace cur table with a placeholder
+    # - Detect S3 paths as variables
+    resources['views'] = {}
+    cur_tables = []
+    for key, view_data in all_views_data.items():
+        if all_databases and isinstance(view_data.get('data'), str):
+            view_data['data'] = view_data['data'].replace(f'{all_databases[0]}.', '${athena_database_name}.')
 
-    template_arn = res['Arn']
-    logger.info(f'Template arn = {template_arn}')
+        if isinstance(view_data.get('data'), str):
+            view_data['data'] = view_data['data'].replace('CREATE VIEW ', 'CREATE OR REPLACE VIEW ')
 
-    time.sleep(5)
+        # Analyse dependancies: if the dependancy is CUR there is a special flag
+        deps = view_data.get('dependsOn', {})
+        non_cur_dep_views = []
+        for dep_view in deps.get('views', []):
+            if cur_helper.table_is_cur(name=dep_view):
+                logger.debug(f'{dep_view} is cur')
+                view_data['dependsOn']['cur'] = True
+                # replace cur table name with a variable
+                if isinstance(view_data.get('data'), str):
+                    view_data['data'] = view_data['data'].replace(f'{dep_view}', '${cur_table_name}')
```

Comment:
cur tables treated separately as we don't create CUR table, but assume it should be created by customer, right? Could you please add this as a comment

Reply:
Added